xapian-bindings/configure.ac

```diff
@@ -287,11 +291,31 @@ case $with_python$with_python3$with_php7$with_ruby$with_tcl$with_csharp$with_jav
   test -z "$with_ruby" && with_ruby=no
   test -z "$with_perl" && with_perl=no
   test -z "$with_lua" && with_lua=no
+  test -z "$with_go" && with_go=no
   ;;
 esac
 
 BINDINGS=
-
+if test no != "$with_go" ; then
+  AC_PATH_PROG([GO],["go"],[])
+  AC_MSG_CHECKING(go)
+  if test -n "$GO" ; then
+    AC_MSG_RESULT(yes)
+    AC_MSG_CHECKING(goroot)
+    GOROOT=`go env GOROOT`
```

Comment:
You carefully probe for `go` above (which is good) but then hard-code `go` here instead of using the probed value in `$GO`.

Reply:
sir could you please explain little bit in detail..I didn't understand..you mean to use ${GO} in line 305?

Reply:
No need for "sir" here.

Say for example I have the version of go I want to use installed as `/home/olly/private/go/bin/go` - I specify `./configure GO=home/olly/private/go/bin/go` and the probe sets `GO` inside configure to that, but then here you just run `go` which searches `PATH` instead.  If there isn't a `go` on `PATH` this command will fail; if there is, it'll set `GOROOT` to the value for the wrong install of go.

(The user could instead set `PATH` to point first to the `go` you want to use to achieve this effect, but given we're supporting `GO=...` we ought to actually implement it correctly).